src/transformStateWithClones.js

```diff
@@ -7,7 +7,49 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrOfStates = [];
+
+  arrOfStates.push(state);
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      const stateCopy = { ...arrOfStates[arrOfStates.length - 1] };
```

Comment:
`const stateCopy = { ...arrOfStates[arrOfStates.length - 1] };` is used in every condition. DRY - you can move it to the beggining of the loop.